File: Java/DiagonalDifference.java
//The question requires the absolute difference between the sums of a square matrix diagonals
//Solved by BrunoRijo at 08-08-222 22:38

import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {
    
    public static int diagonalDifference(List<List<Integer>> arr) {
    int right = 0, left = 0;
    int count1 = 0, count2 = arr.get(0).size()-1;
    
    while(count1 < arr.get(0).size()){
        right += arr.get(count1).get(count1);
        left += arr.get(count1).get(count2);
        count1++;
        count2--;
    }
    
    return Math.abs(right-left);    
    }    
    
}

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int n = Integer.parseInt(bufferedReader.readLine().trim());

        List<List<Integer>> arr = new ArrayList<>();

        IntStream.range(0, n).forEach(i -> {
            try {
                arr.add(
                    Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
                        .map(Integer::parseInt)
                        .collect(toList())
                );
            } catch (IOException ex) {
                throw new RuntimeException(ex);
            }
        });

        int result = Result.diagonalDifference(arr);

        bufferedWriter.write(String.valueOf(result));
        bufferedWriter.newLine();

        bufferedReader.close();
        bufferedWriter.close();
    }
}

File: Kotlin/DiagonalDifference.kt
import java.io.*
import java.math.*
import java.security.*
import java.text.*
import java.util.*
import java.util.concurrent.*
import java.util.function.*
import java.util.regex.*
import java.util.stream.*
import kotlin.collections.*
import kotlin.comparisons.*
import kotlin.io.*
import kotlin.jvm.*
import kotlin.jvm.functions.*
import kotlin.jvm.internal.*
import kotlin.ranges.*
import kotlin.sequences.*
import kotlin.text.*

fun diagonalDifference(arr: Array<Array<Int>>): Int {
    var right = 0
    var left = 0
    var count1 = 0
    var count2 = arr.get(0).size-1
    while(count1 <= arr.get(0).size-1){
       right += arr.get(count1).get(count1) 
       left += arr.get(count1).get(count2) 
       count1++
       count2--
    }
    return Math.abs(right-left)
    
}

fun main(args: Array<String>) {
    val n = readLine()!!.trim().toInt()

    val arr = Array<Array<Int>>(n, { Array<Int>(n, { 0 }) })

    for (i in 0 until n) {
        arr[i] = readLine()!!.trimEnd().split(" ").map{ it.toInt() }.toTypedArray()
    }

    val result = diagonalDifference(arr)

    println(result)
}

